### ISSUE TITLE
Allow shift down only when tree is rooted

### ISSUE THREAD

**Issue by goodideagiver** [RakambdaOrg/FallingTree#145]
### Describe the feature

In depth explanation:
An option to check if any of the blocks at the very bottom of the tree are connected to dirt/grass or netherrack (nether trees).

Additional options:
1. Set a radius. Example: If the block at the bottom of the tree has dirt/grass/netherrack in a set radius - the tree will be cut
2. Grass like block whitelist (for other mods etc.)

Why this feature?
This would allow the mod to not cut player structures like roofs etc.

**Comment by Rakambda** [comment]
I get the point of this feature. Though this doesn't go well with shift down. If for example you break the tree with the 2nd block from the ground, at some point it won't be touching the ground.

Adding this would just add again another config and make it even more complex than it already is. In my opinion this can be solved with the "adjacent blocks" whitelist option. I'm aware that this is quite tedious if you play with lots of mods, but in a sense it describes what you want. Just that instead of describing ONE REQUIRED block to be around, you list ALL those that CAN be around.

By default this list will contain the logs + leaves + air. If anything else touches the tree, it won't be cut. In a player structure it is very likely that the log part of the build is touching something that won't be in the whitelist.

**Comment by goodideagiver** [comment]
> Adding this would just add again another config and make it even more complex than it already is. In my opinion this can be solved with the "adjacent blocks" whitelist option. I'm aware that this is quite tedious if you play with lots of mods, but in a sense it describes what you want. Just that instead of describing ONE REQUIRED block to be around, you list ALL those that CAN be around.
> 
> By default this list will contain the logs + leaves + air. If anything else touches the tree, it won't be cut. In a player structure it is very likely that the log part of the build is touching something that won't be in the whitelist.

I tried using this list and It didn't allow me to cut jungle trees even when I specified minecraft:vine adjacent block

**Comment by Rakambda** [comment]
You probably also need to add the cocoa beans and grass/dirt. Maybe I can improve that a bit by displaying in the logs what block prevented the tree to be cut.

**Comment by goodideagiver** [comment]
> 
> 
> You probably also need to add the cocoa beans and grass/dirt. Maybe I can improve that a bit by displaying in the logs what block prevented the tree to be cut.

There was no cocoa beans and grass+dirt was also whitelisted

**Comment by Rakambda** [comment]
Do you perhaps have the seed + coordinates so I can see a bit what the tree looks like ? (if you don't have other mods that change world generation tho)

**Comment by goodideagiver** [comment]
It was a regular jungle tree spawned by growing it from 4 saplings. Every other vanilla tree type was working properly.

**Comment by Rakambda** [comment]
Setting minecraft:dirt and minecraft:vine as adjacent blocks worked fine for me after I grew a jungle tree.

You can try with this version (1.17 fabric, as I don't know what version you use), it will log in the console/logs what block prevented the cut: https://www.dropbox.com/t/ij7UpgqFA1gzry81

Eg: `[19:42:02] [Server thread/INFO] (fallingtree) Didn't cut tree at Block{minecraft:oak_log}, reason: Found block Block{minecraft:dirt} that isn't whitelisted`.

**Comment by goodideagiver** [comment]
I couldn't reproduce my first issue because It was on a server so I had to fix it quickly. Now I specified the blocks 
![obraz](https://user-images.githubusercontent.com/55458485/124363229-65427480-dc3a-11eb-82ab-5d6a804e62af.png)
and it works as intended. I think the original issue might have been created by config file not being updated properly on the server.

I tested the jar and the message works. It would be helpful if it showed in the actionbar like this: `/title @s actionbar "Adjacent block stopped tree cutting"`
